packages/esm-patient-registration-app/src/patient-registration/form-manager.ts

```diff
@@ -188,13 +188,38 @@ export default class FormManager {
     return [];
   }
 
+  static checkIfEdited = (obj1, obj2) => {
+    const objectKeys = Object.keys(obj1);
+
+    for (const objKey of objectKeys) {
+      const val1 = obj1[objKey];
+      const val2 = obj2[objKey];
+      const isAnObject = val1 != null && typeof val1 === 'object';
+      if (
+        (isAnObject && Object.keys(val1).length !== Object.keys(val2).length) ||
+        (isAnObject && !FormManager.checkIfEdited(val1, val2)) ||
+        (!isAnObject && val1 !== val2)
+      ) {
+        return false;
+      }
+    }
+
+    return true;
+  };
+
   static getPatientToCreate(
     values: FormValues,
     personAttributeSections: any,
     patientUuidMap: PatientUuidMapType,
     initialAddressFieldValues: Record<string, any>,
     identifiers: Array<PatientIdentifier>,
   ): Patient {
+    let address = FormManager.getPatientAddressField(values, initialAddressFieldValues);
+
+    if (FormManager.checkIfEdited(initialAddressFieldValues, address)) {
+      address = {};
+    }
```

Comment:
Did you try

```suggestion
    if (isEqual(initialAddressFieldValues, address)) {
      address = {};
    }
```

using [lodash `isEqual`](https://lodash.com/docs/4.17.15#isEqual)?

Reply:
Thanks! I will replace the custom function with the 'lodash' function and make another commit.